### PR TITLE
JKNS-267: minimize jenkins e2es (pipeline strategy verification only)

### PR DESCRIPTION
**Pull request by gabemontero**
In stages during 4.9 and 4.10, we duplicated all the current openshift/origin jenkins e2e's with operator style tests in the sync plugin and client plugin respectively.

We then created step-registry items in openshift/release so that the openshift/jenkins repo and login plugin could leverage those sync and client plugin tests.

We then change the PR defs for all 4 jenkins repos (image repo, 3 plugins) to use the new tests and *STOP* using the jenkins e2e's defined here in openshift/origin.

And lastly, we created periodics for the sync and clienit plugin e2e tests so that they are run a couple of times a day against master branch OCP to help catch any regressions caused by OCP (though nothing have that sort has occured in a while).

Our jenkins e2e journey culminates with this PR.  As part of removing Jenkins from the OCP Payload, we are vastly pruning the jenkins tests in openshift/origin to *ONLY* verify that the Jenkins PIpeline Build Strategy (which has been deprecated since 4.1) from OpenShift Builds still works.

We are preserving the multiple jenkins related e2e suites in `openshift-tests` so that an periodics etc. referencing them will still test the meets min verification that "openshift jenkins" works.

### CODE REVIEW

**Comment by coreydaley** [comment]
It looks good to me, but I will defer to @akram since he is more "in the know" about Jenkins stuff.

**Comment by gabemontero** [comment]
api server pain across the board .... for example

`{  fail [github.com/openshift/origin/test/extended/util/client.go:755]: Mar 16 20:36:11.796: Get "https://api.ci-op-7gx0x10g-139dc.aws-2.ci.openshift.org:6443/apis/user.openshift.io/v1/users/e2e-test-jenkins-pipeline-zwz6j-user": dial tcp 100.20.55.173:6443: i/o timeout}`

let's just try the e2e we care about and see if that is less stressful

/test e2e-aws-jenkins

**Comment by gabemontero** [comment]
and e2e-aws-jenkins is green https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26914/pull-ci-openshift-origin-master-e2e-aws-jenkins/1504214288699494400

the test in question took 2 minutes 13 seconds :-)

**Comment by gabemontero** [comment]
ci/install fail ... trying again

/test e2e-gcp-builds

**Comment by gabemontero** [comment]
the e2e-gcp-builds run https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26914/pull-ci-openshift-origin-master-e2e-gcp-builds/1504447252695879680
had some non-sig-builds failiures, but our updated jenkins e2e passed:

```
:   [sig-builds][Feature:JenkinsRHELImagesOnly][Feature:Jenkins][Feature:Builds][sig-devex][Feature:Jenkins][Slow]  openshift pipeline build  jenkins pipeline build config strategy using a  jenkins instance launched with the ephemeral template | 2m21s
-- | --
```

we are good for approval / review for lgtm /retest to get past unrelated flakes

File: test/extended/builds/pipeline_origin_bld.go [hunk 1]
```diff
@@ -1,161 +1,42 @@
...
-	syncLicenseText                      = "About OpenShift Sync"
-	clientPluginName                     = "openshift-client"
-	syncPluginName                       = "openshift-sync"
-	secretName                           = "secret-to-credential"
-	secretCredentialSyncLabel            = "credential.sync.jenkins.openshift.io"
-	envVarsPipelineGitRepoBuildConfig    = "test-build-app-pipeline"
-)
-
-var _ = g.Describe("[sig-builds][Feature:Builds][sig-devex][Feature:Jenkins][Slow] openshift pipeline build", func() {
+var _ = g.Describe("[sig-builds][Feature:JenkinsRHELImagesOnly][Feature:Jenkins][Feature:Builds][sig-devex][Feature:Jenkins][Slow] openshift pipeline build", func() {
```
Comment (gabemontero) [inline test/extended/builds/pipeline_origin_bld.go:15]: combining all the existing jenkins related  `[Feature:...]` from both this file and the 2 pipeline_*.go files that are deleted here means that all the existing `openshift-tests` suites use this same test via the ginkgo searching mechanism

File: test/extended/builds/pipeline_origin_bld.go [hunk 2]
```diff
@@ -1,161 +1,42 @@
...
-		pollingInterval                       = time.Second
-		timeout                               = time.Minute
-		oc                                    = exutil.NewCLI("jenkins-pipeline")
-		ticker                                *time.Ticker
-		j                                     *jenkins.JenkinsRef
-		pvs                                   = []*corev1.PersistentVolume{}
-		nfspod                                = &corev1.Pod{}
+		oc               = exutil.NewCLI("jenkins-pipeline")
+		j                *jenkins.JenkinsRef
+		simplePipelineBC = exutil.FixturePath("testdata", "builds", "simple-pipeline-bc.yaml")
```
Comment (gabemontero) [inline test/extended/builds/pipeline_origin_bld.go:21]: our new super simple jenkins pipeline strategy build config

File: test/extended/builds/pipeline_origin_bld.go [hunk 3]
```diff
@@ -1,161 +1,42 @@
...
-			case useSyncPluginSnapshotImage:
-				fmt.Fprintf(g.GinkgoWriter, "\nTHE UPCOMING TESTS WILL LEVERAGE AN IMAGE THAT EXTENDS THE LATEST OPENSHIFT JENKINS IMAGE AND OVERRIDES THE OPENSHIFT SYNC PLUGIN WITH A NEW VERSION BUILT FROM PROPOSED CHANGES TO THAT PLUGIN.\n")
-				licensePrefix = syncLicenseText
-				pluginName = syncPluginName
-				useSnapshotImage = true
-				newAppArgs = syncPluginNewAppArgs
-			default:
-				fmt.Fprintf(g.GinkgoWriter, "\nNO PR TEST ENV VARS SET SO TESTING CURRENT PLUGIN VERSIONS IN LATEST OPENSHIFT JENKINS IMAGE ON DOCKER HUB.\n")
-			}
+			newAppArgs := []string{"jenkins-ephemeral", "-p", "MEMORY_LIMIT=2Gi", "-p", "DISABLE_ADMINISTRATIVE_MONITORS=true"}
```
Comment (gabemontero) [inline test/extended/builds/pipeline_origin_bld.go:34]: all the new-app related hoopla that has been deleted was related to testing on 3.x

not needed for 4.x

File: test/extended/builds/pipeline_origin_bld.go [hunk 4]
```diff
@@ -215,430 +86,34 @@ var _ = g.Describe("[sig-builds][Feature:Builds][sig-devex][Feature:Jenkins][Slo
...
-					out, err := j.GetJobConsoleLogsAndMatchViaBuildResult(br, "Finished: SUCCESS")
-					o.Expect(err).NotTo(o.HaveOccurred())
-					g.By("making sure job log ran with our config map slave pod template")
-					o.Expect(out).To(o.ContainSubstring("Running on jenkins-slave"))
-
-					g.By("clean up openshift resources for next potential run")
-					err = oc.Run("delete").Args("configmap", "jenkins-slave").Execute()
-					o.Expect(err).NotTo(o.HaveOccurred())
-					err = oc.Run("delete").Args("bc", "openshift-jee-sample").Execute()
+					_, err = j.GetJobConsoleLogsAndMatchViaBuildResult(br, "Finished: SUCCESS")
```
Comment (gabemontero) [inline test/extended/builds/pipeline_origin_bld.go:110]: by the way, this has the additional benefit of verifying the integration with oauth via the login plugin

File: test/extended/testdata/builds/simple-pipeline-bc.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
+metadata:
+  name: minimalpipeline
+spec:
+  strategy:
+    jenkinsPipelineStrategy:
+      jenkinsfile: |-
+        node() {
+          sh 'exit 0'
```
Comment (gabemontero) [inline test/extended/testdata/builds/simple-pipeline-bc.yaml:10]: can't get any simpler :-)

File: test/extended/util/jenkins/monitor.go [hunk 6]
```diff
@@ -74,62 +65,3 @@ func (jmon *JobMon) Await(timeout time.Duration) error {
 	})
 	return err
 }
-
-func StartJenkinsGCTracking(oc *exutil.CLI, jenkinsNamespace string) *time.Ticker {
```
Comment (gabemontero) [inline test/extended/util/jenkins/monitor.go:78]: removing code that was no longer called by tests

**Comment by gabemontero** [comment]
all green tests @adambkaplan 

PTAL (including some of my "explanation" comments") and let's see if we can get this ready for merge

**Comment by akram** [comment]
all looks good to me @gabemontero 
Including the explanations and the comments. Only a `nit` thing, one comment that could be laid on 2 lines instead of 3.
But, that does not deserve to re-run all the tests for that. Let's save the planet a bit.

/lgtm

**Review by adambkaplan — COMMENTED** [review]
Just the one nit regarding the test name, otherwise I think this looks good.

File: test/extended/testdata/builds/simple-pipeline-bc.yaml (hunk 5 above)
Comment (gabemontero) [reply test/extended/testdata/builds/simple-pipeline-bc.yaml:10]: the latter - executing in primary jenkins pod ("computer" in jenkins parlance)

so we do not need any of the sidecar parlance

File: test/extended/builds/pipeline_origin_bld.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabemontero) [reply test/extended/builds/pipeline_origin_bld.go:15]: ahhh ... now we have to re-run all those green tests :-(

will update this shortly

**Comment by gabemontero** [comment]
weird my answer to your question @adambkaplan shows up in the file diff tab but not the conversation tab

just in case, I think the link to the comment is https://github.com/openshift/origin/pull/26914/files#r831379920

File: test/extended/builds/pipeline_origin_bld.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabemontero) [reply test/extended/builds/pipeline_origin_bld.go]: update pushed @adambkaplan (including `make update` output)

**Comment by gabemontero** [comment]
ok e2e-aws-jenkins is green ... still executing our one test in under 3 minutes

/test e2e-gcp-builds

**Comment by gabemontero** [comment]
e2e-gcp-builds is green again as well !

**Comment by gabemontero** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26914/pull-ci-openshift-origin-master-e2e-aws-jenkins/1505974376204341248 and https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26914/pull-ci-openshift-origin-master-e2e-gcp-builds/1505995431501369344

same execution time for our new test in e2e-gcp-builds as in e2e-aws-jenkins

**Comment by gabemontero** [comment]
all green e2e's @adambkaplan @akram 

shall we tag this for merge?

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by gabemontero** [comment]
unrelated etcd flake

/retest

**Comment by gabemontero** [comment]
same etcd ... showing up in search ci

/retest

**Pull request merged**